src/layout_objects.py

```diff
@@ -5,64 +5,64 @@
 
 from .utils import get_image_size
 from .defines import DEFAULT_SIZE, TagType
-from typing import Dict
+from typing import Any, Dict
+from dataclasses import dataclass
 
-class Pop:
-    def __init__(self, tag: TagType):
-        self.type = TagType.POP
-        self.target = tag
+
+# Represents the interface for grouping the structural objects
+class LPTag:
+    pass
+
+
+@dataclass
+class Pop(LPTag):
+    target: TagType
+    type: TagType = TagType.POP
 
     def __repr__(self):
         return f'<Pop {self.target}>'
 
 
-class LPMeme:
-    def __init__(self, image: str = None, size=None, fillcolor='white', position=None, mode="resize"):
-        self.image = image
-        self.size = size or (None, None)
-        self.fillcolor = fillcolor
-        self.gridposition = position
-        self.mode = mode
-        self.type = TagType.MEME
+@dataclass
+class LPMeme(LPTag):
+    image: str | None = None
+    size: tuple[int | None, int | None] = (None, None)
```

Comment:
I feel like there's been a bit of an implicit change here, allowing more readily for `size = (500, None)`.
Do we want to explicitly support that elsewhere in logic? (e.g. the repr)

Reply:
not really? I think I just wanted to be able to unpack it safely

Reply:
or at the very least, another follow-up